File: macos/Pulse/Models/ServerModel.swift
import Foundation
import SwiftUI

// MARK: - Server Environment

public enum ServerEnvironment: String, Codable, CaseIterable, Identifiable, Sendable {
    case production  = "Production"
    case staging     = "Staging"
    case development = "Development"
    case untagged    = "Other"

    public var id: String { rawValue }

    public var color: Color {
        switch self {
        case .production:  return .red
        case .staging:     return .orange
        case .development: return .blue
        case .untagged:    return .secondary
        }
    }

    public var icon: String {
        switch self {
        case .production:  return "flame.fill"
        case .staging:     return "hammer.fill"
        case .development: return "wrench.and.screwdriver.fill"
        case .untagged:    return "server.rack"
        }
    }

    public var shortLabel: String {
        switch self {
        case .production:  return "prod"
        case .staging:     return "staging"
        case .development: return "dev"
        case .untagged:    return ""
        }
    }
}

// MARK: - ServerModel

public struct ServerModel: Identifiable, Codable, Hashable, Sendable {
    public let id: UUID
    public var name: String
    public var address: String
    public var port: Int
    public var agentID: String?
    public var createdAt: Date
    public var environment: ServerEnvironment
    public var tags: [String]

    public init(
        id: UUID = UUID(),
        name: String,
        address: String,
        port: Int = 8443,
        agentID: String? = nil,
        createdAt: Date = Date(),
        environment: ServerEnvironment = .untagged,
        tags: [String] = []
    ) {
        self.id = id
        self.name = name
        self.address = address
        self.port = port
        self.agentID = agentID
        self.createdAt = createdAt
        self.environment = environment
        self.tags = tags
    }

    public var hostAndPort: String {
        "\(address):\(port)"
    }
}

File: windows/src/Pulse/Core/Models/ServerModel.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Pulse.Core.Models;

public enum ServerEnvironment
{
    Production,
    Staging,
    Development,
    Untagged
}

public class ServerModel
{
    [JsonPropertyName("id")]
    public Guid Id { get; set; } = Guid.NewGuid();

    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("host")]
    public string Host { get; set; } = string.Empty;

    [JsonPropertyName("port")]
    public int Port { get; set; } = 9443;

    [JsonPropertyName("auth_token")]
    public string AuthToken { get; set; } = string.Empty;

    [JsonPropertyName("environment")]
    public ServerEnvironment Environment { get; set; } = ServerEnvironment.Untagged;

    [JsonPropertyName("tags")]
    public List<string> Tags { get; set; } = new();

    [JsonPropertyName("use_tls")]
    public bool UseTls { get; set; } = true;

    [JsonPropertyName("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public string DisplayAddress => $"{Host}:{Port}";
}
